gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/remote/RemoteConnection.java

```diff
@@ -40,6 +40,16 @@
     public static final String GREMLIN_REMOTE = "gremlin.remote.";
     public static final String GREMLIN_REMOTE_CONNECTION_CLASS = GREMLIN_REMOTE + "remoteConnectionClass";
 
+    /**
+     * Key for configuring a per-request timeout option for a {@code RemoteConnection} .
+     */
+    public static final String PER_REQUEST_TIMEOUT = GREMLIN_REMOTE + "timeout";
```

Comment:
@newkek  you know what? i should get rid of this all together. users should just set these parameters the same way they do if they used the `RequestMessage` counterparts. in other words, there shouldn't be "counterparts" - it should just be the options the driver implementation allows. good?

Reply:
Hm, but then unifying those means it imposes a dependency on `gremlin-driver` to be able to parse/use the options, right?

To specify the option in any traversal (not necessarily on a remote traversal connected with the `gremlin-driver`) you need a dependency to `gremlin-driver`

Reply:
yeah - you would need `gremlin-driver`. it feels wrong to me to put them in `gremlin-core` because they don't really apply to `RemoteConnection` (which could be anything) - they apply to `gremlin-driver` and `gremlin-server` which is where all aspects of the protocol live. Putting them in `gremlin-core` feels like we're saying that these options are features of all `RemoteConnection`  and that's not really the intention. 

> To specify the option in any traversal (not necessarily on a remote traversal connected with the gremlin-driver) you need a dependency to gremlin-driver

I know it would be nice if there was a unified list of tokens that could be used for every driver implementation, but I dont' think that was the point of `RemoteConnection`. The idea was that each driver could supply it's own way of doing things and that feels like it extends to these sorts of configuration options (even if they are the same configuration options). I guess if you were a user and you knew that certain config options were the same then you could just treat the config as a string key and then it would work without any additional dependency.......

Reply:
> I guess if you were a user and you knew that certain config options were the same then you could just treat the config as a string key and then it would work without any additional dependency.......

Yes, the point of that comment was that you'd want to use the official Tinkerpop name for the options to parse them. Falling back to using equivalent string values does not protect from if these values change in a Tinkerpop update. But I understand your other point for locating that stuff explicitly in `gremlin-driver`/`gremlin-server`